src/cargo/core/shell.rs

```diff
@@ -182,30 +179,42 @@ impl Shell {
     }
 
     fn fg(&mut self, color: color::Color) -> io::Result<bool> {
-        match self.terminal {
-            Colored(ref mut c) => c.fg(color),
-            NoColor(_) => Ok(false)
+        let color_config = self.config.tty && Auto == self.config.color_config
+            || Always == self.config.color_config;
```

Comment:
Can you refactor this logic into a helper function? That way it can just be:

```rust
match self.terminal {
    Colored(ref mut c) if self.colored() => c.fg(color),
    _ => Ok(false)
}
```